### DISCUSSION TITLE
PINN Configuration - lang

### DISCUSSION THREAD

**Discussion by dshuman52** [procount/pinn#855] [General]
These observations were made when using a RasPI 3 and PINN versions first runs 3.9.2 and later 3.9.3 (last run for sure).

This appears to be more of an annoyance considering that based on my current observation does not adversely affect PINN's intended functionality.

PINN has a short entry area at the bottom of the screen with dropdown list boxes for lang and keyboard.  When lang is changed from "English (UK)" to "English (US)" the keyboard value changes from "gb" to "us" (nice in my case).  However after a reboot the lang value ("not sticky") reverts to "English(UK)" and the keyboard value is still "us" ("sticky").  

I tried to fix this using the cmdline.txt; but the documentation is conflicting in this area at one point it reads:

> lang=<two-letter language code> (e.g. lang=de or lang=en)

shortly below that is says:

> lang=\<country code\>: Sets the language to be used by PINN

From prior experience I expect lang to be something like 'en_GB.UTF-8' or 'en_US.UTF-8'

NOTE: Reviewing cmdline.txt it appears the sticky keyboard parameter was not stored in that file.

I tried modifying cmdline.txt after the first boot to include 'lang=en', 'lang=us' and 'lang="en_US.UTF-8"' I believe I also tried it without the double quotes and all had no effect.

Starting over this morning I tried editing cmdline.txt before the first PINN boot for fun.  I added 'lang="en_US.UTF-8" keyboard=us' and more unrelated options.  The keyboard=us worked replacing 'gb' for me.  The lang= option did NOT work; still have "English (UK)".  Reviewing the cmdline.txt file the options got shuffled in the text after the first boot but neither lang or keyboard were dropped.  One other changed option probably misspelled by me was dropped (not an issue , my fault) during the reshuffle.

Again this does not appear to affect the functionality of PINN,

**Comment by procount** [comment]
Although the keyboard changes between US and GB, there is no difference between the US and GB languages in PINN - they are the same. So trying to force it to US is largely futile. This is a hangover from the original NOOBS program.

Is there any text within PINN that would benefit from being Americanised?

**Comment by dshuman52** [comment]
As stated this is probably more an annoyance than a problem.  I was not sure.

    On Saturday, December 14, 2024 at 04:00:06 PM EST, procount ***@***.***> wrote:  
 
 


Although the keyboard changes between US and GB, there is no difference between the US and GB languages in PINN - they are the same. So trying to force it to US is largely futile. This is a hangover from the original NOOBS program.

Is there any text within PINN that would benefit from being Americanised?
